src/OpenTelemetry.Exporter.Zipkin/Implementation/ZipkinConversionExtensions.cs

```diff
@@ -0,0 +1,187 @@
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Diagnostics;
+using OpenTelemetry.Trace;
+using OpenTelemetry.Trace.Export;
+
+namespace OpenTelemetry.Exporter.Zipkin.Implementation
+{
+    internal static class ZipkinConversionExtensions
+    {
+        private const string StatusCode = "ot.status_code";
+        private const string StatusDescription = "ot.status_description";
+
+        private static readonly Dictionary<string, int> RemoteEndpointServiceNameKeyResolutionDictionary = new Dictionary<string, int>(StringComparer.OrdinalIgnoreCase)
+        {
+            ["net.peer.name"] = 0, // RemoteEndpoint.ServiceName primary.
+            ["peer.service"] = 0, // RemoteEndpoint.ServiceName primary.
+            ["peer.hostname"] = 1, // RemoteEndpoint.ServiceName alternative.
+            ["peer.address"] = 1, // RemoteEndpoint.ServiceName alternative.
+            ["http.host"] = 2, // RemoteEndpoint.ServiceName for Http.
+            ["db.instance"] = 2, // RemoteEndpoint.ServiceName for Redis.
+        };
+
+        private static readonly ConcurrentDictionary<string, ZipkinEndpoint> LocalEndpointCache = new ConcurrentDictionary<string, ZipkinEndpoint>();
+        private static readonly ConcurrentDictionary<string, ZipkinEndpoint> RemoteEndpointCache = new ConcurrentDictionary<string, ZipkinEndpoint>();
+
+        internal static ZipkinSpan ToZipkinSpan(this SpanData otelSpan, ZipkinEndpoint defaultLocalEndpoint, bool useShortTraceIds = false)
+        {
+            var context = otelSpan.Context;
+            var startTimestamp = ToEpochMicroseconds(otelSpan.StartTimestamp);
+            var endTimestamp = ToEpochMicroseconds(otelSpan.EndTimestamp);
+
+            var spanBuilder =
+                ZipkinSpan.NewBuilder()
+                    .TraceId(EncodeTraceId(context.TraceId, useShortTraceIds))
+                    .Id(EncodeSpanId(context.SpanId))
+                    .Kind(ToSpanKind(otelSpan))
+                    .Name(otelSpan.Name)
+                    .Timestamp(ToEpochMicroseconds(otelSpan.StartTimestamp))
+                    .Duration(endTimestamp - startTimestamp);
+
+            if (otelSpan.ParentSpanId != default)
+            {
+                spanBuilder.ParentId(EncodeSpanId(otelSpan.ParentSpanId));
+            }
+
+            Tuple<string, int> remoteEndpointServiceName = null;
+            foreach (var label in otelSpan.Attributes)
+            {
+                string key = label.Key;
+                string strVal = label.Value.ToString();
+
+                if (strVal != null
+                    && RemoteEndpointServiceNameKeyResolutionDictionary.TryGetValue(key, out int priority)
+                    && (remoteEndpointServiceName == null || remoteEndpointServiceName.Item2 > priority))
```

Comment:
I think this should be `<` as the priority is in ascending order from 0 - we want the lowest ranked peer.service.

Reply:
It actually works, there's a unit test for it. But something about the way it is written is confusing. I flipped it around I think it reads better now. LMK